src/main/java/com/conveyal/gtfs/stats/StopStats.java

```diff
@@ -1,7 +1,239 @@
+                        StopTime st = feed.stop_times.get(t.b);
+                        Route route = feed.routes.get(trip.route_id);
+
+                        List<StopTime> times = new ArrayList<>();
+                        if (routeStopTimeMap.containsKey(route.route_id)) {
+                            times.addAll(routeStopTimeMap.get(route.route_id));
+                        }
+                        times.add(st);
+                        routeStopTimeMap.put(route.route_id, times);
+                    }
+                });
+        Map<Fun.Tuple2<String, String>, TIntList> waitTimesByRoute = new HashMap<>();
+        Map<Fun.Tuple2<String, String>, Set<Fun.Tuple2<StopTime, StopTime>>> missedTransfers = new HashMap<>();
+        // iterate over every entry in route -> list of stopTimes map
+        for (Map.Entry<String, List<StopTime>> entry : routeStopTimeMap.entrySet()) {
+            final int MISSED_TRANSFER_THRESHOLD = 60 * 10;
+            List<StopTime> currentTimes = entry.getValue();
+
+            String currentRoute = entry.getKey();
+
+            for (StopTime currentTime : currentTimes) {
+                if (currentTime.arrival_time > 0) {
+                    // cycle through all other routes that stop here.
+                    for (Map.Entry<String, List<StopTime>> entry2 : routeStopTimeMap.entrySet()) {
+
+                        List<StopTime> compareTimes = entry2.getValue();
+                        String compareRoute = entry2.getKey();
+                        Fun.Tuple2<String, String> routeKey = new Fun.Tuple2(currentRoute, compareRoute);
+                        if (compareRoute.equals(currentRoute)) {
+                            continue;
+                        }
+                        if (!waitTimesByRoute.containsKey(routeKey)) {
+                            waitTimesByRoute.put(routeKey, new TIntArrayList());
+                        }
+
+                        int shortestWait = Integer.MAX_VALUE;
+                        // compare current time against departure times for route
+                        for (StopTime compareTime : compareTimes) {
+                            if (compareTime.departure_time > 0) {
+                                int waitTime = compareTime.departure_time - currentTime.arrival_time;
+
+                                // if non-negative and shortest, save for later
+                                if (waitTime >= 0 && waitTime < shortestWait){
+                                    shortestWait = waitTime;
+                                }
+                                // otherwise, check for missed near-transfer opportunities
+                                else {
+                                    if (waitTime < 0 && waitTime * -1 <= MISSED_TRANSFER_THRESHOLD) {
+                                        Fun.Tuple2<StopTime, StopTime> missedTransfer = new Fun.Tuple2(compareTime, currentTime);
```

Comment:
is that enough info?  seems like it might be useful to have at least trip ID associated with the times.